### PR TITLE
feat(zimbra): #MC-265 edit external calendar lightbox

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
new external calendar lightbox with zimbra option

## Checklist tests
click on "add external calendar button"

## Issue ticket number and link
[MC-265](https://jira.support-ent.fr/browse/MC-265)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts [hunk 1]
```diff
@@ -29,6 +35,8 @@ class Controller implements IViewModel {
     }
 
     $onInit() {
+        console.log("testt");
```
Comment (c4role) [inline src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts:38]: As-tu besoin de garder ce console.log ?

**Review by c4role — APPROVED** [review]

File: src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.html [hunk 2]
```diff
@@ -3,43 +3,79 @@
...
+                    <div class="twelve cell vertical-spacing-four">
+                        <i18n>calendar.addition.choice</i18n> :
+                    </div>
+
+
+                    <div class="">
+                        <div class="row" >
+                            <input class="cell right-spacing-small top-spacing" ng-if="vm.enableZimbra" type="radio"
+                                   ng-click="vm.changeSelection('icsLink')" name="calendarOrigin" checked/>
+                            <label class="cell"><i18n>calendar.external.from.url</i18n> : </label>
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.html:25]: tu peux utiliser des html entities plutôt

File: src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts [hunk 3]
```diff
@@ -50,11 +57,19 @@ class Controller implements IViewModel {
...
     isFormComplete(): boolean {
-        return !!this.$scope.vm.calendar && !!this.$scope.vm.calendar.title && !!this.$scope.vm.calendar.icsLink;
+        return !!this.$scope.vm.calendar && !!this.$scope.vm.calendar.title && (!!this.$scope.vm.calendar.icsLink !== !!this.$scope.vm.calendar.platform);
     }
 
-    translate(text: string): string {
-        return lang.translate(text);
+
+    changeSelection(field: string, platform?: string): void {
+        switch (field) {
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts:65]: this.$scope.vm.calendar[field] = undefined;

ça prend moins de place 😉

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts (hunk 3 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/directives/external-calendar-form/external-calendar-form.ts:65]: vu avec @qpic-cgi

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**